uiowa/uiowa01
# Feature: Responsive view mode options for image embeds

# How to Test
__Note:__ _Testing this is a bit involved and a little hard to explain. Its going to take a little trial and error to get the hang of the process._
1. Read the section of this article describing how to use the browser to test different device settings, which starts just before the linked anchor: https://freshman.tech/responsive-images/#responsive-images-in-html
2. `drush @theming.local si collegiate`
3. Create a piece of content
4. In the body field, add an image for each type of view mode. There are 12 of them. It helps to use a different image for each one, for reasons you'll see later.
5. For each image you added, click the "Edit media" button to set the view mode. Set one to each of the options. Again, there should be 12.
6. View the page in a browser with Responsive Design Mode turned on (Ctrl-Shift-M in Chrome, Cmd-Option-M in Firefox).
7. Based on the [rules outlined for each image size](https://github.com/uiowa/uiowa01/issues/221#issuecomment-564691800) (Full, Large, Medium, and Small), determine whether the images that are being loaded correspond to the viewport size you are testing against. It's probably easiest if you just ask me how to do this when you are ready.
8. Repeat this at different viewport sizes and with different Device Pixel Ratios.

Please report back any issues you run into as well as your thoughts about how well we are providing coverage for different devices and whether we need to offer all of these options.
